src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,951 @@
+        /// 
+        /// public int GetF(int x, int y) // Generated method
+        /// {
+        ///     return x * y;
+        /// }
+        /// 
+        /// public void M(int x, int y, int f)
+        /// {
+        ///     Console.WriteLine(f);
+        /// }
+        /// 
+        /// public void InvokeMethod()
+        /// {
+        ///     M(5, 6, GetF(5, 6)); //Fills in with call to generated method
+        /// }
+        /// 
+        /// -----------------------------------------------------------------------
+        /// ****Overload Example:****
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = [|x * y|];
+        ///     Console.WriteLine(f);
+        /// }
+        /// 
+        /// public void InvokeMethod()
+        /// {
+        ///     M(5, 6);
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public void M(int x, int y) // Generated overload
+        /// {
+        ///     M(x, y, x * y)
+        /// }
+        /// 
+        /// public void M(int x, int y, int f)
+        /// {
+        ///     Console.WriteLine(f);
+        /// }
+        /// 
+        /// public void InvokeMethod()
+        /// {
+        ///     M(5, 6);
+        /// }
+        /// </summary>
+        private async Task<SyntaxNode> ModifyDocumentInvocationsTrampolineOverloadAndIntroduceParameterAsync(
+            Compilation compilation, Document currentDocument, Document originalDocument,
+            List<SyntaxNode> invocations, Dictionary<TIdentifierNameSyntax, IParameterSymbol> mappingDictionary,
+            IMethodSymbol methodSymbol, SyntaxNode containingMethod,
+            bool allOccurrences, string parameterName, TExpressionSyntax expression, bool trampoline, bool overload,
+            CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(currentDocument);
+            var semanticFacts = currentDocument.GetRequiredLanguageService<ISemanticFactsService>();
+            var invocationSemanticModel = await currentDocument.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var root = await currentDocument.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var editor = new SyntaxEditor(root, generator);
+            var syntaxFacts = currentDocument.GetRequiredLanguageService<ISyntaxFactsService>();
+            var insertionIndex = GetInsertionIndex(compilation, methodSymbol, syntaxFacts, containingMethod);
+
+            // Creating a new method name by concatenating the parameter name that has been camelcased.
+            var newMethodIdentifier = "Get" + parameterName.First().ToString().ToUpper() + parameterName[1..];
+
+            var validParameters = methodSymbol.Parameters.AsEnumerable().SelectMany(parameter => mappingDictionary.Values.Where(param => parameter.Equals(param))).ToImmutableArray();
```

Comment:
AsEnumerable should not be necessry.

Reply:
i don't understand this.  `mappingDictionary.Values.Where(param => parameter.Equals(param)))` is jsut asking if the param is in the values of that map. a nd if so, it jsut selects hte param... multiple times.

I think this is just `methodSymbol.Parameters.Intersect(mappingDictionary.Values)`

Reply:
Yeah I was just trying to maintain the original order of the parameters, didnt see the intersect method